app/models/grda_warehouse/hud/assessment.rb

```diff
@@ -75,8 +75,8 @@ def question_matching_requirement(question, answer = nil, case_sensitive: true)
       return nil if matching_question.blank?
       return matching_question unless answer.present?
 
-      assessment_answer = matching_question.AssessmentAnswer.to_s
-      assessment_answer.downcase! unless case_sensitive
+      assessment_answer = matching_question.AssessmentAnswer&.to_s
+      assessment_answer&.downcase! unless case_sensitive
       answer&.downcase! unless case_sensitive
 
       assessment_answer == answer
```

Comment:
I want to check this more, because this might evaluate true with `nil == nil` , which it wasn't doing before. I need to understand the context better.

Reply:
`nil.to_s` produces an empty string, so how about
```
assessment_answer = matching_question.AssessmentAnswer&.to_s || ''
```

Reply:
that gave a lint warning, I used `(assessment_answer || '') == answer`